### PR TITLE
Add support for Azure BYOK exocompute

### PR DESCRIPTION
**Pull request by johan3141592**

### CODE REVIEW

File: examples/aws_exocompute/main.go [hunk 1]
```diff
@@ -76,25 +79,25 @@ func main() {
 		fmt.Printf("Feature: %v, Regions: %v, Status: %v\n", feature.Name, feature.Regions, feature.Status)
 	}
 
-	// Add exocompute config for the account.
-	exoID, err := awsClient.AddExocomputeConfig(ctx, aws.ID(aws.Default()),
-		aws.Managed("us-east-2", "vpc-4859acb9", []string{"subnet-ea67b67b", "subnet-ea43ec78"}))
+	// Add an exocompute configuration for the account.
+	exoID, err := exoClient.AddAWSConfiguration(ctx, accountID,
+		exocompute.AWSManaged(gqlaws.RegionUsEast2, "vpc-4859acb9", []string{"subnet-ea67b67b", "subnet-ea43ec78"}))
```
Comment (Draper1) [inline examples/aws_exocompute/main.go:84]: Is there a reason for using hard-coded VPC and Subnet IDs for AWS?

File: examples/azure_exocompute/main.go [hunk 2]
```diff
@@ -82,25 +85,25 @@ func main() {
 		fmt.Printf("Feature: %v, Regions: %v, Status: %v\n", feature.Name, feature.Regions, feature.Status)
 	}
 
-	// Add exocompute config for the account.
-	exoID, err := azureClient.AddExocomputeConfig(ctx, azure.CloudAccountID(accountID),
-		azure.Managed("eastus2", "/subscriptions/9318aeec-d357-11eb-9b37-5f4e9f79db5d/resourceGroups/terraform-test/providers/Microsoft.Network/virtualNetworks/terraform-test/subnets/default"))
+	// Add an exocompute configuration to the cloud account.
+	configID, err := exoClient.AddAzureConfiguration(ctx, accountID,
+		exocompute.AzureManaged(gqlazure.RegionEastUS2, "/subscriptions/9318aeec-d357-11eb-9b37-5f4e9f79db5d/resourceGroups/terraform-test/providers/Microsoft.Network/virtualNetworks/terraform-test/subnets/default"))
```
Comment (Draper1) [inline examples/azure_exocompute/main.go:90]: Same question applies here regarding hard-coded values?

File: examples/aws_exocompute/main.go (hunk 1 above)
Comment (johan3141592) [reply examples/aws_exocompute/main.go:84]: They are made up placeholders for the example code. Anybody wanting to run the code has to put in their own valid values.

File: examples/azure_exocompute/main.go (hunk 2 above)
Comment (johan3141592) [reply examples/azure_exocompute/main.go:90]: They are made up placeholders for the example code. Anybody wanting to run the code has to put in their own valid values.

File: pkg/polaris/graphql/azure/cloud.go [hunk 3]
```diff
@@ -353,10 +361,15 @@ func (a API) UpgradeCloudAccountPermissionsWithoutOAuth(ctx context.Context, id
 	a.log.Print(log.Trace)
 
 	query := upgradeAzureCloudAccountPermissionsWithoutOauthQuery
+	var queryFeature any = feature.Name
```
Comment (Draper1) [inline pkg/polaris/graphql/azure/cloud.go:364]: Is the name ever expected to not be a string? should we type this to string instead?

**Review by Draper1 — APPROVED** [review]
Couple small comments but otherwise LGTM.

File: pkg/polaris/graphql/azure/cloud.go (hunk 3 above)
Comment (johan3141592) [reply pkg/polaris/graphql/azure/cloud.go:364]: Name is always a string. However, if the feature has permission groups (`len(feature.PermissionGroups) > 0`), we have to use a different query. The other query expects a feature and not a string, using `any` here means both values can be assigned to same variable.

**Pull request merged**